pkg/transport/listener.go

```diff
@@ -269,3 +270,27 @@ func ShallowCopyTLSConfig(cfg *tls.Config) *tls.Config {
 	}
 	return &ncfg
 }
+
+// IsClosedConnError returns true if the error is from closing listener, cmux.
+// copied from golang.org/x/net/http2/http2.go
+func IsClosedConnError(err error) bool {
+	if err == nil {
+		return false
+	}
+
+	// TODO: remove this string search once go/http2 has better type handling
+	operr, ok := err.(*net.OpError)
+	if (ok && operr.Err.Error() == "use of closed network connection") ||
+		strings.Contains(err.Error(), "use of closed network connection") {
+		return true
+	}
+
+	// TODO: match '/go/internal/poll.ErrClosing'
+	if strings.Contains(err.Error(), "use of closed file or network connection") {
+		return true
+	}
+
+	// TODO: match exact error value 'cmux.ErrListenerClosed'
+	// for now just string match without dependency
+	return err.Error() == "mux: listener closed"
+}
```

Comment:
would it be enough have:
```go
func IsClosedConnectionError(err error) bool {
    return err != nil && strings.Contains(err.Error(), "closed")
}
```
or is there some case where it has `closed` in the error string but the connection isn't closed?